docs/rules/Proj0046.md

```diff
@@ -0,0 +1,39 @@
+---
+parent: General
+ancestor: MSBuild
+---
+
+# Proj0046: Update statements should change state
+In a MSBuild file, `<ItemGroup>` items can be updated using the `@Update`
+attribute. This rule checks if the update changes the state of the item by
+defining another attribute or child elements.
+
+## Non-compliant
+``` xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net9.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Update="Qowaiv" />
+  </ItemGroup>
+
+</Project>
+```
+
+## Compliant
+``` xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net9.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Update="Qowaiv" Version="8.4.3" />
```

Comment:
This should only be compliant if there is a PackageReference Include="Qowaiv" elsewhere

Reply:
@Gemberkoekje That might be true, but that should be another rule (of which I was thinking).